### ISSUE TITLE
ERM should also save best model based on worst-group accuracy

### ISSUE THREAD

**Issue by Haoxiang-Wang** [kohpangwei/group_DRO#10]
Hi @kohpangwei and @ssagawa ,

Your paper mentioned that _"All benchmark models were evaluated at the best early stopping epoch (as measured by robust validation accuracy)."_ However, your code
https://github.com/kohpangwei/group_DRO/blob/ca58872bd5a7b5fe90c35d4c39504babe76e3532/train.py#L205-L209
indicates that (i) for ERM, the best model is determined by the average validation accuracy (ii) for reweighting and GroupDRO, the best model is determined by the worst-group accuracy. I think for a fair comparison, the model selection rule of (i) should be changed to be identical to (ii), what do you think?

Did you use the model selection rule (i) for ERM in your paper's experiments (e.g., Table 3)? I'm trying to reproduce your results, but I'm not sure if your results on ERM are from (i) or not.

**Comment by kohpangwei** [comment]
Hi Haoxiang,

You're right that train.py saves the best ERM model by average validation accuracy. However, the results in our paper are all reported (as the paper says) on early stopping on robust validation accuracy. The analysis script here https://worksheets.codalab.org/bundles/0x701597e91c7f4ce5a6896428500f3535 calls `print_accs` which uses `robust_acc` for early stopping: https://github.com/kohpangwei/group_DRO/blob/master/analysis_utils.py#L121

Hope that helps and thanks for checking!
